curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-string-and-array-methods-by-building-a-music-player/653635c731206b718659d3d5.md

```diff
@@ -0,0 +1,640 @@
+---
+id: 653635c731206b718659d3d5
+title: Step 22
+challengeType: 0
+dashedName: step-22
+---
+
+# --description--
+
+Inside the `playSong` function, set the `audio.src` property equal to `song.src`. This tells the audio element where to find the audio data for the selected song.
+
+Also, set the `audio.title` property equal to `song.title`. This tells the audio element what to display as the title of the song.
+
+# --hints--
+
+Test 1
+
+```js
+
+```
```

Comment:
The tests:

You should not modify the exist `playSong` function and its content.

```js
assert.match(code, /const\s*playSong\s*=\s*\(id\)\s*=>\s*\{\s*const\s*song\s*=\s*userData\?\.songs\.find\(\(song\)\s=>\s*song.id\s===\s*id\);?/)
```

You should set `audio.src` to `song.src`.

```js
assert.match(code, /const\s*playSong\s*=\s*\(id\)\s*=>\s*\{\s*const\s*song\s*=\s*userData\?\.songs\.find\(\(song\)\s=>\s*song.id\s===\s*id\);?\s*audio\.src\s*=\s*song\.src;?/)
```

You should set `audio.title` to `song.title`.

```js
assert.match(code, /const\s*playSong\s*=\s*\(id\)\s*=>\s*\{\s*const\s*song\s*=\s*userData\?\.songs\.find\(\(song\)\s=>\s*song.id\s===\s*id\);?\s*audio\.src\s*=\s*song\.src;?\s*audio\.title\s*=\s*song\.title;?\s*\};?/)
```